docs/how_tos/theming.md

```diff
@@ -0,0 +1,204 @@
+
+By serving CSS loaded externally, consuming applications of Paragon no longer need to be responsible for compiling the theme SCSS to CSS themselves and instead use a pre-compiled CSS file. In doing so, this allows making changes to the Paragon theme without needing to necessarily re-build and re-deploy all consuming applications.
+
+### Dark mode and theme variant preferences
+
+`@openedx/frontend-platform` supports both `light` (required) and `dark` (optional) theme variants. The choice of which theme variant should be applied on page load is based on the following preference cascade:
+
+1. **Get theme preference from localStorage.** Supports persisting and loading the user's preference for their selected theme variant, until cleared.
+1. **Detect user system settings.** Rely on the `prefers-color-scheme` media query to detect if the user's system indicates a preference for dark mode. If so, use the default dark theme variant, if one is configured.
+1. **Use default theme variant as configured (see below).** Otherwise, load the default theme variant as configured by the `defaults` option described below.
+
+Whenever the current theme variant changes, an attrivbute `data-paragon-theme-variant="*"` is updated on the `<html>` element. This attribute enables applications' both JS and CSS to have knowledge of the currently applied theme variant.
+
+### Supporting custom theme variants beyond `light` and `dark`
+
+If your use case necessitates additional variants beyond the default supported `light` and `dark` theme variants, you may pass any number of custom theme variants. Custom theme variants will work the user's persisted localStorage setting (i.e., if a user switches to a custom theme variant, the MFE will continue to load the custom theme variant by default). By supporting custom theme variants, it also supports having multiple or alternative `light` and/or `dark` theme variants.
+
+### Performance implications
+
+There is also a meaningful improvement in performance as loading the compiled theme CSS from an external CDN means micro-frontends (MFEs) can include cached styles instead of needing to load essentially duplicate theme styles included in each individual MFE as users navigate across the platform.
+
+However, as the styles from `@openedx/paragon` and `@openedx/brand-openedx` get loaded at runtime by `@openedx/frontend-platform`, the associated CSS files do not get processed through the consuming application's Webpack build process (e.g., if the MFE used PurgeCSS or any custom PostCSS plugins specifically for Paragon).
+
+### Falling back to styles installed in consuming application
+
+If any of the configured external `PARAGON_THEME_URLS` fail to load for whatever reason (e.g., CDN is down, URL is incorrectly configured), `@openedx/paragon` will attempt to fallback to the relevant files installed in `node_modules` from the consuming application.
```

Comment:
@PKulkoRaccoonGang  thanks for the feedback, the fallback url was missing the `publicPath` I solved it.

![image](https://github.com/openedx/frontend-platform/assets/66016493/6a070ff5-ea66-4cda-a313-156a816bbe46)

Reply:
We don’t receive `undefined`, but I couldn’t load styles from the locally installed Paragon package in `node_modules`. Could you check if `getParagonThemeCss` works correctly in a PR related to a [frontend-build](https://github.com/openedx/frontend-build/pull/546)?

<img width="1492" alt="image" src="https://github.com/openedx/frontend-platform/assets/93188219/c07a5684-df1b-4a20-9e5f-f8427b7b9d49">

Reply:
The problem here is related to tutor-mfe, they define `BASE_URL` as the `MFE_HOST` ([you can see here](https://github.com/overhangio/tutor-mfe/blob/master/tutormfe/patches/openedx-lms-development-settings#L4)) thing that is not 100% true in dev mode because the URL needs the `PORT` that is specific for each MFE

You can  add in the settings:

``` python
MFE_CONFIG_OVERRIDES = {
     "discussions": {
        "BASE_URL": "apps.local.edly.io:2002"
    }
}
```